### PR TITLE
fix(node:tls): set TLSSocket.alpnProtocol for client connections

### PR DESCRIPTION
**Pull request by satyarohith**
Towards https://github.com/denoland/deno/issues/26127

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Looks fine for me. @littledivy any comments?

File: ext/node/polyfills/_tls_wrap.ts [hunk 1]
```diff
@@ -151,10 +153,20 @@ export class TLSSocket extends net.Socket {
...
+          try {
+            const hs = await conn.handshake();
+            if (hs.alpnProtocol) {
+              tlssock.alpnProtocol = hs.alpnProtocol;
+            } else {
+              tlssock.alpnProtocol = false;
+            }
+          } catch {
+            // Don't interrupt "secure" event if handshake fails
+          }
```
Comment (littledivy) [inline ext/node/polyfills/_tls_wrap.ts:165]: Might want to emit an error event here. Can you check what Node.js does?

Comment (satyarohith) [reply ext/node/polyfills/_tls_wrap.ts:165]: The error event is emitted in the read/write operation.

**Review by littledivy — APPROVED** [review]
LGTM

**Pull request merged**